pkg/eval/sys/sys.go

```diff
@@ -0,0 +1,25 @@
+// Package sys exposes variables and functions that deal with the specific
+// system being run on. Such as the OS name and CPU architecture.
+package sys
+
+import (
+	"runtime"
+
+	"github.com/elves/elvish/pkg/eval"
+	"github.com/elves/elvish/pkg/eval/vars"
+)
+
+//elvdoc:var arch
+//
+// The architecture of the platform; e.g., amd64, arm, ppc. This is read-only.
+
+//elvdoc:var platform
+//
+// The name of the platform; e.g., darwin (macOS), linux. This is read-only.
+
+var Ns = eval.NewNs()
+
+func init() {
+	Ns.Add("arch", vars.NewReadOnly(runtime.GOARCH))
+	Ns.Add("platform", vars.NewReadOnly(runtime.GOOS))
```

Comment:
I imagine you called it after Python's `sys.platform`, but I'd rather this be called `os` to align with Go.

Reply:
I did but on second thought agree with using `os` for consistency given how tightly Elvish is bound to Go.

Reply:
I also assume you'll want the namespace changed from `sys` to `os`.

Reply:
And the straightforward rename exposes a problem: 

```
pkg/program/shell/runtime.go:18:2: os redeclared as imported package name
        previous declaration at pkg/program/shell/runtime.go:7:2
```

Which means that any import of the elvish os module must rename it. Which means we now need to establish a convention for this situation. 😦 I have tentatively settled on importing `elvish_os "github.com/elves/elvish/pkg/eval/os"`.

Reply:
My preference is calling this module `sys`; it is a bit awkward to have a variable called `$os:os`, no?

Reply:
Note that my first incarnation did call this the `sys:` namespace :smile: Not sure what you mean by your `$os:os` example. There is no such var at this time. There is `$os:platform` and `$os:arch`. I deliberately used `platform` because `os` has too much baggage. For example, why is the value `darwin` rather than `macos` on my Macs? Or `linux` rather than `ubuntu`? Etcetera. I think `platform` better conveys the meaning of this var.